rfcs/053-filtering-by-subject/README.md

```diff
@@ -0,0 +1,84 @@
+# RFC 053: Filtering by contributor, genre, and subject
+
+For per-subject and per-person pages, we want to filter for images and works that match a given subject/person.
+For example:
+
+*   Works by Florence Nightingale
+*   Works about Charles Darwin
+*   Images about mental health
+
+This turns out to be non-trivial, so this RFC describes how we'll get there.
+
+Strictly speaking we only need subjects and contributors for this work, but genres are so similar we should treat them in the same way.
+
+## Requirements
+
+1.  On per-concept pages, there's a sample of matching images/works.
+
+2.  On per-concept pages, there's a link to a filtered search for the given concept, directly below the sample results.
+
+3.  On work pages, the list of subjects/contributors/genres to link to filtered searches for each subject.
+
+4.  In the works API, there are filters and aggregations for subject/contributor/genre.
+
+5.  Given a single work in the works API, there should be an obvious way to construct a filter URL for works with the same subjects/contributors/genres as this work.
+
+6.  The catalogues are the source of truth for subject identifiers.
+    We can find equivalent identifiers, but we can't pick them from scratch.
+
+    e.g. If a Sierra record has a subject tagged with an LCSH identifier, we can find the Wikidata subject with that identifier.
+    If a Sierra record has a subject with no identifier, we can't choose an identifier, even if we could find a Wikidata subject with a matching label.
+
+## Current behaviour
+
+We have filters and aggregations for *label*, not ID.
+
+## Future behaviour
+
+If we add filtering/aggregations for subjects by ID, we already know how they'll be named: `subjects` and `source.subjects` for works and images, respectively.
+This is consistent with our existing API design.
+
+But do we add filtering/aggregations for subjects by ID?
+We find ourselves in a dilemma:
+
+*   We can't rely on filtering by label, because there may be concepts with similar/identical labels but which refer to different things.
+    e.g. two members of the same family.
+
+*   We can't rely on filtering by ID, because not all subjects/genres/contributors are identified.
+
+From an API perspective, we could easily support both, but it's more complicated in the front-end.
+How do we choose which filter mechanism to use?
+How does an API client choose which filter mechanism to use?
+
+Questions:
+
+*   Are the requirements as stated correct?
```

Comment:
A general response to the problem stated: it seems to me that the shortcomings of each approach (ID vs label) are kind of related to each other. That is to say - I think the existence of unidentifiable concepts in some sense implies an inability to disambiguate concepts by label. 

In reality it's slightly more complex than that - as you say, with family names it's realistic to end up with multiple ambiguous _and_ identified concepts. Still, I'm not sure there's really a problem per se...

> 1. On per-concept pages, there's a sample of matching images/works.

Concept pages are for identified concepts, this is fine with the ID search

> 1. On per-concept pages, there's a link to a filtered search for the given concept, directly below the sample results.

I'm not sure what this requirement means - is a "search for the given concept" a works/image search? Again, I feel that it's fine for this only to be for identified concepts.

> 3. On work pages, the list of subjects/contributors/genres to link to filtered searches for each subject.

I wonder if an approach here would be to link to a label search for an unidentified concept, and an ID search for an identified concept. Perhaps there could be UI signifiers that we were doing a text search rather than searching for some kind of canonical entity.

> 4. In the works API, there are filters and aggregations for subject/contributor/genre.

Think this is implied by the above 3 requirements.

> 5. Given a single work in the works API, there should be an obvious way to construct a filter URL for works with the same subjects/contributors/genres as this work.

As with 3 but agreed it isn't necessarily obvious. 

> 6. The catalogues are the source of truth for subject identifiers.

For sure.

**Appendix**

I think I might have talked myself into (via responses to 3 and 5) some kind of concept filter that accepts IDs _and_ labels? Is that an unacceptable deviation? It would behave much like the existing works search when searching for an ID.

Reply:
I suppose it would be better named a "concepts query" rather than a "concepts filter" in this case. I'm going off it already tbh - I'm not sure how the aggregations would work, although I think that would be the expected behaviour in this case.

Re search for unidentified concept labels returning identified concepts, I think that's fine.

Reply:
I agree with the general gist of what Jamie is saying above. Concept pages are for identified concepts, if it's not identified we don't give it a page. We tend towards more identified concepts over time (and it is the vast majority already, genres aside). For genres that means, no concept pages but a continuation of what we have today.

The caveat to the above is the potential mixing of user flows. If all genres do one thing and all subjects do another, that's probably ok. If it's less deterministic than that we may have a problem.

I would like @GarethOrmerod to be involved in this conversation, as I think we probably do need to treat linking to concept pages and back to searches for labels differently when we link to them from works pages, to make sure it's clear to user where's they're going to go in each case.